src/repo.rs

```diff
@@ -60,6 +60,23 @@ impl Repository {
         }
     }
 
+    /// Find and open an existing repository, respecting git environment
+    /// variables.  This acts like `open_ext` with the
+    /// `REPOSITORY_OPEN_FROM_ENV` flag, but additionally respects `$GIT_DIR`.
+    /// With `$GIT_DIR` unset, this will search for a repository starting in
+    /// the current directory.
```

Comment:
Sure, I can elaborate a bit.  I'd rather not duplicate the entire libgit2 documentation for this, though.

(Not least of which because git2-rs uses a different license than libgit2, so I can't just copy/paste from libgit2.)